src/modeling_library/switch/update.jl

```diff
@@ -0,0 +1,70 @@
+mutable struct SwitchUpdateState{T}
+    weight::Float64
+    score::Float64
+    noise::Float64
+    prev_trace::Trace
+    trace::Trace
+    index::Int
+    discard::ChoiceMap
+    updated_retdiff::Diff
+    SwitchUpdateState{T}(weight::Float64, score::Float64, noise::Float64, prev_trace::Trace) where T = new{T}(weight, score, noise, prev_trace)
+end
+
+function process!(gen_fn::Switch{C, N, K, T},
```

Comment:
@marcoct This seems correct for the "namespace merging" semantics. In `update`, if I switch branches, I generate with a merge of the previous traces choices and the choice map. I'm worried this will throw an error (however) if not all constraints are visited (e.g. if the previous trace has non-empty set difference with the new namespace).

I'm also dispatching on `process!` using the diff types. I'm not sure if this current implementation is current - mostly because I'm not supported `Diffed` dispatch yet (and I suspect I have to).

Reply:
> I'm also dispatching on process! using the diff types. I'm not sure if this current implementation is correct - mostly because I'm not supported Diffed dispatch yet (and I suspect I have to).

@femtomc The `Diffed` types are only used for boxing Julia values for diff propagation of Julia code via operator overloading. GFI implementers only need to worry about `Diff` values. So what you have is right.